src/lib/db/user-store.ts

```diff
@@ -53,27 +54,57 @@ const rowToUser = row => {
     });
 };
 
+export interface IUserLookup {
+    id?: number;
+    username?: string;
+    email?: string;
+}
+
+export interface IUserSearch {
+    name?: string;
+    username?: string;
+    email: string;
+}
+
+export interface IUserUpdateFields {
+    name?: string;
+    email?: string;
+}
+
 class UserStore {
-    constructor(db, getLogger) {
+    private db: Knex;
+
+    private logger: Logger;
+
+    constructor(db: Knex, getLogger: LogProvider) {
         this.db = db;
         this.logger = getLogger('user-store.js');
     }
 
-    async update(id, user) {
+    async update(id: number, fields: IUserUpdateFields): Promise<User> {
         await this.db(TABLE)
             .where('id', id)
-            .update(mapUserToColumns(user));
+            .update(mapUserToColumns(fields));
         return this.get({ id });
     }
 
-    async insert(user) {
+    async insert(user: User): Promise<User> {
```

Comment:
I see you only add types here, so pointing out that we should probably just use `returning` to return the user directly instead of performing one more query to get the user is more a nit than a request for change as part of this PR

Reply:
Agree, that is a good improvement.